korrecte-lib/src/linters/lints/alb_ingress_instance.rs

```diff
@@ -0,0 +1,187 @@
+        }
+
+        let services = Self::extract_service_names(ingress);
+        let ingress_type = IngressType::from(ingress);
+        let missconfigured_services = self.check_allowed_service_type(&ingress_type, services);
+
+        for service in missconfigured_services {
+            let finding = Finding::new(Self::spec(), ingress.meta().clone())
+                .add_metadata("service", service.meta().name.clone());
+
+            reporter.report(finding);
+        }
+    }
+}
+
+impl<'a> AlbIngressInstance<'a> {
+    fn spec() -> LintSpec {
+        LintSpec {
+            group: Group::Configuration,
+            name: "alb_ingress_controller_instance_missconfiguration".to_string(),
+        }
+    }
+}
+
+impl<'a> AlbIngressInstance<'a> {
+    fn check_allowed_service_type(
+        &self,
+        ingress_type: &IngressType,
+        service_names: HashSet<String>,
+    ) -> Vec<&Box<Object<ServiceSpec, ServiceStatus>>> {
+        self.object_repository
+            .all()
+            .iter()
+            .filter_map(Self::filter_service)
+            .filter(|service| {
+                Self::is_service_missconfigured(service, ingress_type, &service_names)
+            })
+            .collect()
+    }
+
+    fn filter_service(object: &KubeObjectType) -> Option<&Box<Object<ServiceSpec, ServiceStatus>>> {
+        match object {
+            KubeObjectType::V1Service(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    fn is_service_missconfigured(
+        service: &Box<Object<ServiceSpec, ServiceStatus>>,
+        ingress_type: &IngressType,
+        service_names: &HashSet<String>,
+    ) -> bool {
+        let default_service_type = "clusterip".to_string();
+
+        if !service_names.contains(&service.meta().name) {
+            return false;
+        }
+
+        let service_type = service.spec.type_.as_ref().unwrap_or(&default_service_type);
+        !ingress_type.is_service_type_allowed(&service_type)
+    }
+
+    fn extract_service_names(ingress: &Object<IngressSpec, IngressStatus>) -> HashSet<String> {
+        let empty = Vec::new();
+        let services: HashSet<String> = ingress
+            .spec
+            .rules
+            .as_ref()
+            .unwrap_or(&empty)
+            .iter()
+            .flat_map(|rule| {
+                rule.http.as_ref().map(|http| {
+                    http.paths
+                        .iter()
+                        .map(|path| path.backend.service_name.clone())
+                        .collect()
+                })
+            })
+            .collect();
+
+        services
+    }
+}
+
+enum IngressType {
+    Instance,
+    Ip,
+    Other,
+}
+
+impl From<&Object<IngressSpec, IngressStatus>> for IngressType {
+    fn from(ingress: &Object<IngressSpec, IngressStatus>) -> Self {
+        match ingress
+            .meta()
+            .annotations
+            .get("alb.ingress.kubernetes.io/target-type")
+            .map(|ty| ty.as_str())
```

Comment:
What does "ty" mean in this context? type?

Reply:
Yes, changed it for `target_type`